Add colorable berets to the loadout menu

SierraKomodo opened the pull request:
Also fixed a small bug with the white beret sprite that had a random transparent pixel that wasn't supposed to be transparent. Some screen caps of the beret in action:

![loadoutmenu](https://user-images.githubusercontent.com/11140088/28845778-fcbd2fee-76bd-11e7-9d80-ac44178e2e28.PNG)

![green](https://user-images.githubusercontent.com/11140088/28845785-fff9b952-76bd-11e7-91f8-97af048958e7.PNG)

![peach](https://user-images.githubusercontent.com/11140088/28845787-01b56200-76be-11e7-9eff-68d735db1e97.PNG)

![white](https://user-images.githubusercontent.com/11140088/28845790-035c3c50-76be-11e7-8d3a-8b8c87c8e2e5.PNG)

![stuff](https://user-images.githubusercontent.com/11140088/28845906-66e0bc92-76be-11e7-97e6-2f2dfddab106.PNG)

mustafakalash commented:
berets never looked good on any of the hairstyles ive ever picked

SierraKomodo commented:
And for the record, yes, I did include job restrictions to `NON_MILITARY_ROLES`

AtomicWorm commented:
This could use a squash.

Techhead0 commented:
Oh jeez. I didn't even notice the commit count. Yeah, please squash this if you can.

mustafakalash commented:
tbh squash and merge works perfectly fine

SierraKomodo commented:
I have no idea what a squash is or how to do it.

SierraKomodo commented:
I _think_ I did this right. But to be honest, I'm not sure, because it's still showing all of the commits on github while gitkraken shows the branch as not having them anymore.

SierraKomodo commented:
Yeeeah, that apparently didn't do what you wanted at all, despite it saying 'squash'. So, I have no idea how to squash commits.

SierraKomodo committed:
Add colored berets to loadout

Add object definition for plain color beret


Add loadout entry for plain color beret


Restrict berets to non military roles


Move beret loadout definition to shared file


Beret loadout is properly defined now. Hopefully.


Use plain white beret sprite


Fix transparent pixel in white beret sprite


Merge remote-tracking branch 'origin/features/loadout-berets' into features/loadout-berets

SierraKomodo commented:
All fixed.
